pkg/controller/shared/legacy_kube_informers.go

```diff
@@ -337,3 +339,115 @@ func (f *namespaceInformer) Lister() *cache.IndexerToNamespaceLister {
 }
+
+type LimitRangeInformer interface {
+	Informer() framework.SharedIndexInformer
+	Indexer() cache.Indexer
+	Lister() StoreToLimitRangeLister
+}
+
+type limitRangeInformer struct {
+	*sharedInformerFactory
+}
+
+func (f *limitRangeInformer) Informer() framework.SharedIndexInformer {
+	f.lock.Lock()
+	defer f.lock.Unlock()
+
+	informerObj := &kapi.LimitRange{}
+	informerType := reflect.TypeOf(informerObj)
+	informer, exists := f.informers[informerType]
+	if exists {
+		return informer
+	}
+
+	lw := f.customListerWatchers.GetListerWatcher(kapi.Resource("limitrange"))
+	if lw == nil {
+		lw = &cache.ListWatch{
+			ListFunc: func(options kapi.ListOptions) (runtime.Object, error) {
+				return f.kubeClient.LimitRanges(kapi.NamespaceAll).List(options)
+			},
+			WatchFunc: func(options kapi.ListOptions) (watch.Interface, error) {
+				return f.kubeClient.LimitRanges(kapi.NamespaceAll).Watch(options)
+			},
+		}
+	}
+
+	informer = framework.NewSharedIndexInformer(
+		lw,
+		informerObj,
+		f.defaultResync,
+		cache.Indexers{},
+	)
+	f.informers[informerType] = informer
+
+	return informer
+}
+
+func (f *limitRangeInformer) Indexer() cache.Indexer {
+	informer := f.Informer()
+	return informer.GetIndexer()
+}
+
+func (f *limitRangeInformer) Lister() StoreToLimitRangeLister {
+	informer := f.Informer()
+	return StoreToLimitRangeLister{Indexer: informer.GetIndexer()}
+}
+
+// StoreToReplicationControllerLister gives a store List and Exists methods. The store must contain only LimitRanges.
+type StoreToLimitRangeLister struct {
+	cache.Indexer
+}
+
+func (s StoreToLimitRangeLister) LimitRanges(namespace string) storeLimitRangesNamespacer {
+	return storeLimitRangesNamespacer{s.Indexer, namespace}
+}
+
+type storeLimitRangesNamespacer struct {
+	indexer   cache.Indexer
+	namespace string
+}
+
+func (s storeLimitRangesNamespacer) List(selector labels.Selector) ([]*kapi.LimitRange, error) {
+	var controllers []*kapi.LimitRange
+
+	if s.namespace == kapi.NamespaceAll {
+		for _, m := range s.indexer.List() {
+			rc := m.(*kapi.LimitRange)
+			if selector.Matches(labels.Set(rc.Labels)) {
+				controllers = append(controllers, rc)
+			}
+		}
+		return controllers, nil
+	}
+
+	key := &kapi.LimitRange{ObjectMeta: kapi.ObjectMeta{Namespace: s.namespace}}
+	items, err := s.indexer.Index(cache.NamespaceIndex, key)
+	if err != nil {
+		for _, m := range s.indexer.List() {
+			rc := m.(*kapi.LimitRange)
+			if s.namespace == rc.Namespace && selector.Matches(labels.Set(rc.Labels)) {
+				controllers = append(controllers, rc)
+			}
+		}
+		return controllers, nil
+	}
+	for _, m := range items {
+		rc := m.(*kapi.LimitRange)
+		if selector.Matches(labels.Set(rc.Labels)) {
+			controllers = append(controllers, rc)
+		}
+	}
+	return controllers, nil
+}
+
+func (s storeLimitRangesNamespacer) Get(name string) (*kapi.LimitRange, error) {
+	obj, exists, err := s.indexer.GetByKey(s.namespace + "/" + name)
```

Comment:
limit ranges are namespaced, so every extant limitrange would be stored by a "ns/name" key